PR: Improve content and organization of history pane in docs (#211)
spyder-ide/spyder-docs: doc/historylog.rst

History
#######

With the **History** pane, you can view all the commands you've entered into any :doc:`ipythonconsole`, along with their timestamp.

.. image:: images/history/history-standard.png
   :alt: Spyder History Log, displaying a list of previously executed commands

======================
Using the History pane
======================

Navigating the :guilabel:`History` pane is very straightforward.
Each Spyder session is marked by a date and timestamp, making it easy to remember when you executed a certain command.
Statements can be selected and copied from the context menu or with the normal system shortcuts.
Just like in the Editor, selecting a word or phrase displays all other occurrences, and full syntax highlighting is also supported.

============
Options Menu
============

The top-right options menu (:guilabel:`Hamburger` icon) allows you to toggle wrapping of long lines (:guilabel:`Wrap lines`), and whether the line number is displayed to the left of the text (:guilabel:`Show line numbers`).

> .. image:: images/history/history-wrap.gif

Reviewer: I think this gif should be easier to understand by activating first `Show line numbers` and then `Wrap lines`.

Reviewer 2: Agreed. Otherwise, LGTM.

Reviewer 2: This was addressed

   :alt: Spyder History Log, displaying wraping lines and showing line number

> ==============

Reviewer 2: Remember to fix on rebase

Advanced usage
==============

The list of commands shown in the :guilabel:`History` pane are stored in :file:`history.py` in the :file:`.spyder-py3` directory in your user home folder (by default, :file:`C:/Users/{username}` on Windows, :file:`/Users/{username}` for macOS, and typically :file:`/home/{username}` on GNU/Linux).
You might need to show invisible files in order to see it on a non-Windows operating system.

> .. image:: images/history/history-log-file.png

Reviewer: Could you use the light macOS theme? The dark theme is non-standard and a bit hard to read.

Reviewer 2: - [ ] Its an OS, not Spyder level setting on a minority OS, and likely a majority of developers prefer the dark theme, so I don't see the "non-standard" nature of the dark theme (built-in to the OS and heavily promoted by Apple) as an issue. However, I do agree the text for invisible items is quite hard to read with it enabled, so I guess its worth changing for this particular screenshot.
- [ ] It would be much better to show the file in a real editor with syntax highlighting, so its clearer its just a normal Python file rather than some bespoke text format.
- [ ] The size of the screenshot, at nearly 2 MB, is gargantuan for what it shows, larger than most GIFs. Downscaling by 2x and applying `optipng` results in a size of only 0.35 MB, less than 1/5 of the original, for essentially no loss in perceptible quality at the final displayed resolution. Again, to make things simplest, I can do this as part of the rebase pass once you've updated the screenshot accordingly.

Reviewer: > Its an OS, not Spyder level setting on a minority OS, and likely a majority of developers prefer the dark theme, so I don't see the "non-standard" nature of the dark theme

I disagree. It took me a long time (i.e. several seconds) to understand what that screenshot was about. I didn't get that a Finder instance was displayed at the left until I paid close attention to the window the details.

Also, please remember that our docs are for people that are just getting started with Spyder. Of course it can be useful for experienced developers too, but that's not our main target.

Finally, macOS can have a small market share but most tutorials and videos are done with it. So I think most people are accustomed to see Mac screenshots here and there.

> However, I do agree the text for invisible items is quite hard to read with it enabled, so I guess its worth changing for this particular screenshot.

Yep, that's another important issue.

> It would be much better to show the file in a real editor with syntax highlighting

Agreed. You can use Sublime for that @juanis2112.

> The size of the screenshot, at nearly 2 MB, is gargantuan for what it shows

Good point too.

Reviewer 2: > I disagree. It took me a long time (i.e. several seconds) to understand what that screenshot was about. I didn't get that a Finder instance was displayed at the left until I paid close attention to the window the details.

Okay, then if it is an issue with familiarity, that makes sense then. Thanks for explaining.

Reviewer 2: Its using a real editor now, but the session timestamp (`############ Date ##########` that the history log adds) is not visible anymore, so its not clear its a history log file. I would suggest showing that. Also, it looks much clearer to me what's going on, though as far as I can tell, it looks like its still using the dark theme so I'm not sure if @ccordoba12 still thinks it should be changed (but I would assume yes).

Reviewer: > I would suggest showing that

Agreed.

> I'm not sure if @ccordoba12 still thinks it should be changed (but I would assume yes).

Correct, I still prefer to see Finder on the left with the Mac light theme.

Reviewer 2: @juanis2112 Should be last two things on this PR, other than the line break and image stuff I'll fix:

* [ ] Scroll editor to the top of the file to show the session timestamp, etc.
* [ ] Use light OS theme

Reviewer 2: LGTM now.

   :alt: Spyder History Log file

> 

Reviewer 2: Reminder to remove this spurious line break on rebase

History
#######

With the **History** pane, you can view all the commands entered into any :doc:`ipythonconsole` along with their time-stamp.

> .. image:: images/history/history-menu.png

Reviewer 2: Per @juanis2112 :

> Replace screenshot for screenshot without the options menu

   :alt: Spyder History Log, displaying a list of previously executed commands

> 
> (Replace screenshot for screenshot without the options menu)

Reviewer 2: ```suggestion
```

* ~Grammar: "for" -> "with"~
* Unless its a placeholder for an image that isn't there at all, I would suggest that you use the OP, Github comments or elsewhere for meta remarks like this, so they don't pollute the repo itself and cause unnecessary problems when rebasing. Thanks.

> =================
> Using the History
> =================

Reviewer 2: Needs "pane" here to be clear that its referring to the pane, rather than the history shown by the pane, a usage which matches e.g. the Find pane docs.

```suggestion
======================
Using the History pane
======================
```

Navigating the :guilabel:`History` pane is very straightforward.

> Each Spyder session is marked by a date and time-stamp, making it easy to remember when you executed a certain command.

Reviewer 2: ```suggestion
Each Spyder session is marked by a date and timestamp, making it easy to remember when you executed a certain command.
```

My mistake

Statements can be selected and copied from the context menu or with the normal system shortcuts.

> Just like in the editor, highlighting a word or phrase displays all other occurrences, and full syntax highlighting is also supported.

Reviewer 2: ```suggestion
Just like in the Editor, selecting a word or phrase displays all other occurrences, and full syntax highlighting is also supported.
```

Avoid repetition in my original text

============
Options Menu
============

> The top-right options menu (:guilabel:`Gear` icon) allows you to set the number of commands the :guilabel:`History` pane should remember.

Reviewer 2: ```suggestion

The top-right options menu (:guilabel:`Hamburger` icon) allows you to set the number of commands that the :guilabel:`History` pane should remember.
```

Needs an update for Spyder 4, a line break and a small grammar tweak

(Gif with dialog for changing the maximum entries)

> You can also toggle wrapping of long lines (:guilabel:`Wrap lines`), and decide whether to display the line number at the left of the pane.

Reviewer 2: ```suggestion
You can also toggle wrapping of long lines (:guilabel:`Wrap lines`), and whether the line number is displayed to the left of the text (:guilabel:`Show line numbers`).
```

Simplify syntax, avoid ambiguity and be consistent about stating the guilabel name,

(Gif to display deactivating wrapping and displaying line number)

==============
Advanced usage
==============

> The commands from the :guilabel:`History` are stored in the :file:`.spyder-py3` directory in your user home folder (by default, :file:`C:/Users/{username}` on Windows, :file:`/Users/{username}` for macOS, and typically :file:`/home/{username}` on GNU/Linux).

Reviewer 2: ```suggestion
The list of commands shown in the :guilabel:`History` pane are stored in :file:`history.py` in the :file:`.spyder-py3` directory in your user home folder (by default, :file:`C:/Users/{username}` on Windows, :file:`/Users/{username}` for macOS, and typically :file:`/home/{username}` on GNU/Linux).
```

Add a few details, and be more clear

You might need to show invisible files in order to see it on a non-Windows operating system.

> (Screenshot of the history log file)

Reviewer 2: Is it really necessary to show this? Its just the same file that's shown in the history log.

Reviewer 2: I guess...but what's the practical use case? Would anyone likely actually use this?

.. image:: images/history/history-menu.png
   :alt: Spyder History Log, displaying a list of previously executed commands

(Replace screenshot for screenshot without the options menu)

======================
Using the History pane
======================

Navigating the :guilabel:`History` pane is very straightforward.
Each Spyder session is marked by a date and timestamp, making it easy to remember when you executed a certain command.
Statements can be selected and copied from the context menu or with the normal system shortcuts.
Just like in the Editor, selecting a word or phrase displays all other occurrences, and full syntax highlighting is also supported.

============
Options Menu
============

The top-right options menu (:guilabel:`Hamburger` icon) allows you to set the number of commands that the :guilabel:`History` pane should remember.

(Gif with dialog for changing the maximum entries)

You can also toggle wrapping of long lines (:guilabel:`Wrap lines`), and whether the line number is displayed to the left of the text (:guilabel:`Show line numbers`).

(Gif to display deactivating wrapping and displaying line number)

==============
Advanced usage
==============

> The list of commands shown in the :guilabel:`History` pane are stored in :file:`history.py` in the :file:`.spyder-py3` directory in your user home folder (by default, :file:`C:/Users/{username}` on Windows, :file:`/Users/{username}` for macOS, and typically :file:`/home/{username}` on GNU/Linux).

Reviewer 2: Using `/` for paths consistently across platforms is what is specified in the style guide and implemented across the current docs for both [paths in general](https://github.com/spyder-ide/spyder-docs/blob/master/STYLEGUIDE.md#writing-all-languages) and the [the file role specifically](https://github.com/spyder-ide/spyder-docs/blob/master/STYLEGUIDE.md#rest-style-elements). Given most paths mentioned in the docs are not platform-specific, it makes sense to just pick one path separator and use it consistently, rather than treating specially the small number of absolute paths that are unique to Windows.

Furthermore, in Python source code `/` is used almost universally for paths regardless of platform (by convention with string paths, and by requirement with `pathlib` `Path`s); indeed, our own copy relative/absolute path commands copy them this way, so it will avoid things breaking things if users copy them into their code, the console. Finally, given this usage, users are users across platforms are sufficiently used to `/` in paths in Python code, URLs and elsewhere that the likelihood of confusion by using `/` on Windows is effectively zero; despite many places in Spyder's UI where `/` is or was used on Windows, the only cases where this has genuinely caused an issue for even beginners was when `/` and `\` were mixed in the _same_ path or otherwise used inconsistently, which this would help avoid.

> The list of commands shown in the :guilabel:`History` pane are stored in :file:`history.py` in the :file:`.spyder-py3` directory in your user home folder (by default, :file:`C:/Users/{username}` on Windows, :file:`/Users/{username}` for macOS, and typically :file:`/home/{username}` on GNU/Linux).

Reviewer: > Using / for paths consistently across platforms is what is specified in the style guide

Ok, I wasn't aware of that. Then we can leave it as it is.

With the **History** pane, you can view all the commands you've entered into any :doc:`ipythonconsole`, along with their timestamp.

> .. image:: images/history/history-standard.png

Reviewer 2: This screenshot looks good, but `optipng` reduces the size of this 0.5 MB image by nearly 50%, entirely for free (aside from a few seconds of runtime). I can do it while rebasing before the merge, if desired.

Reviewer 2: Will be addressed on rebase

The top-right options menu (:guilabel:`Hamburger` icon) allows you to toggle wrapping of long lines (:guilabel:`Wrap lines`), and whether the line number is displayed to the left of the text (:guilabel:`Show line numbers`).

.. image:: images/history/history-wrap.gif
   :alt: Spyder History Log, displaying wraping lines and showing line number

> ==============

Reviewer 2: A line break was mistakenly removed here (3 before second-level headers).

The list of commands shown in the :guilabel:`History` pane are stored in :file:`history.py` in the :file:`.spyder-py3` directory in your user home folder (by default, :file:`C:/Users/{username}` on Windows, :file:`/Users/{username}` for macOS, and typically :file:`/home/{username}` on GNU/Linux).
You might need to show invisible files in order to see it on a non-Windows operating system.

.. image:: images/history/history-log-file.png
   :alt: Spyder History Log file

> 

Reviewer 2: A spurious extra line break was added here.
